CHANGELOG.md

```diff
@@ -11,17 +11,36 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Run name printed at the end of a run so it's easier to find.
 - Type information added to package data. See [PEP 561](https://www.python.org/dev/peps/pep-0561) for more information.
+- A new integration, `transformers`, with two new steps for running seq2seq models.
+- Added `logging_tqdm`, if you don't want a progress bar, but you still want to see progress in the logs.
+- Added `threaded_generator()`, for wrapping generators so that they run in a separate thread from the generator's consumer.
+- Added a new example for evaluating the T0 model on XSum, a summarization task.
+- Added `MappedSequence` for functionally wrapping sequences.
+- Added `TextFormat`, in case you want to store the output of your steps in raw text instead of JSON.
+- Steps can now list arguments in `SKIP_ID_ARGUMENTS` to indicate that the argument should not affect a step's
+  unique id. This is useful for arguments that affect the execution of a step, but not the output.
+- `Step` now implements `__str__`, so steps look pretty in the debugger.
+- Added `DatasetCombineStep`, a step that combines multiple datasets into one.
 
 ### Fixed
 
 - Fixed torch `StepEarlyCallback` state not being recovered properly on restarts.
+- `LocalWorkspace` now works properly with uncacheable steps.
+- When a Tango run got killed hard, with `kill -9`, or because the machine lost power, `LocalWorkspace` would
+  sometimes keep a step marked as "running", preventing further executions. This still happens sometimes, but it
+  is now much less likely (and Tango gives you instructions for how to fix it).
+- To make all this happen, `LocalWorkspace` now saves step info in a Sqlite database. Unfortunately that means that
```

Comment:
> To make all this happen, `LocalWorkspace` now saves step info in a Sqlite database.

I'm curious how using an Sqlite dataset addresses these issues.

Reply:
We used to store `StepInfo` in the step directory, but uncacheable steps don't have run directories, so we had no way of storing info for uncacheable steps. The web UI was freaking out about that.

It doesn't have to be sqlite, but that was an easy option. We could also create a directory (outside the cache) with a bunch of JSON files, one for each step. Then at least they would be human-readable, which is a plus.

(Sqlite had nothing to do with automatically detecting hung steps, though in real adversarial `kill -9` scenarios, it's probably helpful that it has transactions.)